src/legacybot/units/Archon.java

```diff
@@ -18,35 +27,122 @@ public void run() throws GameActionException {
             // Try/catch blocks stop unhandled exceptions, which cause your robot to explode
             try {
 
-                // Try to shake any trees that we can
-                tryShake(rc);
+                //Start the round
+                roundStart();
 
-                // Generate a random direction
-                Direction dir = randomDirection();
+                // TODO: once we finish implementing state diagrams for every type of unit
+                // We can move this into the start method in Unit.java
+                checkState();
 
-                // Randomly attempt to build a gardener in this direction
-                if (rc.canHireGardener(dir) && Math.random() < .01) {
-                    rc.hireGardener(dir);
+                switch (state){
+                    case SLOW:
+                        slow();
+                        break;
+                    case FAST:
+                        fast();
+                        break;
+                    case AGGRESSIVE:
+                        aggressive();
+                        break;
+                    case INITIAL:
+                        initial();
+                        break;
+                    case DEFENSIVE:
+                        defensive();
+                        break;
                 }
 
-                // Move randomly
-                tryMove(randomDirection(), rc);
-
-                // Broadcast archon's location for other robots on the team to know
-                //MapLocation myLocation = rc.getLocation();
-                //.broadcast(0,(int)myLocation.x);
-                //.broadcast(1,(int)myLocation.y);
-
-                // Try to shake any trees that we can
-                tryShake(rc);
-
-                // Clock.yield() makes the robot wait until the next turn, then it will perform this loop again
-                Clock.yield();
+                // End the round
+                roundEnd();
 
             } catch (Exception e) {
                 System.out.println("Archon Exception");
                 e.printStackTrace();
             }
         }
     }
+
+    public void checkState() throws GameActionException {
+        int round = rc.getRoundNum();
+
+        if(round < 5){
+            state = ArchonUnitState.INITIAL;
+            return;
+        }
+
+        if (round > 2750){
+            state = ArchonUnitState.DEFENSIVE;
+            return;
+        }
+
+        if(round > 2000 || rc.readBroadcast(13) != 0){
+            state = ArchonUnitState.AGGRESSIVE;
+            return;
+        }
+
+        if(round > 1000 || (round / 100 > gardenersBuilt && round > 250)){
+            state = ArchonUnitState.SLOW;
+            return;
+        } else if (round / 100 / totalArchonNum > gardenersBuilt){
+            state = ArchonUnitState.FAST;
+            return;
+        }
```

Comment:
These values could be refactored into constants for better maintainability.

Reply:
I agree, however at this time, I think it would be better if we did that as a separate PR so I can hit some more important issues. I'll add it to the todo list though.